### ISSUE TITLE
font-display:swap - Error. But it`s supported almost all browsers

### ISSUE THREAD

**Issue by schoolphp** [validator/validator#649]
Message: Property font-display doesn't exist.
font-display have good support: https://caniuse.com/#search=font-display

**Comment by sideshowbarker** [comment]
Can you please post a short CSS stylesheet snippet here that uses `font-display`? Either that or a URL for a document online.

I would like to add support for `font-display` but I want to make sure I end up implementing what you actually need.

**Comment by schoolphp** [comment]
https://validator.w3.org/nu/?doc=https%3A%2F%2Fschool-php.com

**Comment by aykut-rocks** [comment]
@font-face {
  font-family: 'XYZ';
  src: url(#) format("woff2"), url(#) format("woff");
  font-weight: 900;
  font-style: normal;
  font-display: fallback;
}

**Comment by schoolphp** [comment]
Main problem of it: 100/100 rating by Google and faster load top of page. 
I have to include into HTML TOP 1000px css with tag <style>. On the top i using fonts, to correctly i have to use "font-display".
Ofcourse i want to show to my clients, that they have 100/100 without any errors site and without font-blinking. (fonts will blinking without using font-display).

**Comment by sideshowbarker** [comment]
I’ve made an interim change to the checker — and pushed it to https://validator.w3.org/nu/ — that causes it to filter out (drop/ignore/suppress) the *Property `font-display` doesn't exist” messages for now (until I can make time to actually implement proper checker support for `font-display`).

**Comment by ylafon** [comment]
Fixed in https://github.com/w3c/css-validator/commit/52af7570bba4196b1468fd6df04c8df31686a159 , 62c4bf2 should be reverted.

**Comment by vielhuber** [comment]
Is this already live? I get some errors with font-display:swap (look at https://validator.w3.org/nu/?doc=https%3A%2F%2Fvielhuber.de%2F).

**Comment by ylafon** [comment]
It is, in your case, you have one instance that is not in a @​font-face rule.

**Comment by vielhuber** [comment]
You are absolutely right, thanks for clarification.

Now I only get one error left, that scrollbar-color is unknown. May I open a github issue for this or is this intentionally that you only accept styles that are not in working draft anymore (https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color)?

**Comment by jaimeiniesta** [comment]
`font-display` is raising `CSS: “font-display”: Property “font-display” doesn't exist.` issues again, what are the plans for this?

Here's an example that triggers this issue:

```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>

<style>
  body{ font-family:Roboto; font-size:15px; font-display:swap;}
</style>

</head>
<body>
<p></p>
</body>
</html>
```

It looks like there's good support for it, and documentation in MDN about this property:

* https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display
* https://caniuse.com/?search=font-display

Should it still be marked as an issue?

**Comment by sideshowbarker** [comment]
> Here's an example that triggers this issue:

That example isn’t valid, so it’s appropriate for the checker to report an error there.

`font-display` isn’t a property, right? It’s a descriptor for use with the `@font-face` at-rule:

https://drafts.csswg.org/css-fonts-4/#font-display-desc

So what would be valid instead is something like this:

```css
@font-face { font-family:Roboto; font-size:15px; font-display:swap; }
```

The checker won’t report any error for that.
